Classes/Emogrifier.php

```diff
@@ -530,6 +530,13 @@ private function fillStyleAttributesWithMergedStyles()
     private function generateStyleStringFromDeclarationsArrays(array $oldStyles, array $newStyles)
     {
         $combinedStyles = array_merge($oldStyles, $newStyles);
+
+        foreach ($oldStyles as $attribute => $expression) {
+            if (isset($newStyles[$attribute]) && strtolower(substr($expression, -10)) === '!important') {
```

Comment:
The string "!important" could be part of a URL or `content` value. It must be at the end. I do need to check for any possible whitespaces though.